src/cloudcasting/utils.py

```diff
@@ -116,3 +119,30 @@ def find_contiguous_t0_time_periods(
     contiguous_time_periods["end_dt"] -= np.timedelta64(forecast_duration)
     assert (contiguous_time_periods["start_dt"] < contiguous_time_periods["end_dt"]).all()
     return contiguous_time_periods
+
+
+# possibly slow (?)
```

Comment:
Possibly worth finding out? :). Could open an issue for this if it's beyond the scope of this PR.